src/foundation.cs

```diff
@@ -8778,8 +9728,107 @@ partial interface NSItemProvider : NSCopying {
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("registerFileRepresentationForTypeIdentifier:fileOptions:visibility:loadHandler:")]
+		void RegisterFileRepresentation (string typeIdentifier, NSItemProviderFileOptions fileOptions, NSItemProviderRepresentationVisibility visibility, RegisterFileRepresentationLoadHandler loadHandler);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("registeredTypeIdentifiersWithFileOptions:")]
+		string[] GetRegisteredTypeIdentifiers (NSItemProviderFileOptions fileOptions);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("hasRepresentationConformingToTypeIdentifier:fileOptions:")]
+		bool HasConformingRepresentation (string typeIdentifier, NSItemProviderFileOptions fileOptions);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Async, Export ("loadDataRepresentationForTypeIdentifier:completionHandler:")]
+		NSProgress LoadDataRepresentation (string typeIdentifier, ItemProviderDataCompletionhandler completionHandler);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Async, Export ("loadFileRepresentationForTypeIdentifier:completionHandler:")]
+		NSProgress LoadFileRepresentation (string typeIdentifier, LoadFileRepresentationHandler completionHandler);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Async (ResultTypeName = "LoadInPlaceResult"), Export ("loadInPlaceFileRepresentationForTypeIdentifier:completionHandler:")]
+		NSProgress LoadInPlaceFileRepresentation (string typeIdentifier, LoadInPlaceFileRepresentationHandler completionHandler);
+
+		[NoWatch, NoTV, NoMac, iOS (11, 0)]
+		[NullAllowed, Export ("suggestedName")]
+		string SuggestedName { get; set; }
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("initWithObject:")]
+		IntPtr Constructor (INSItemProviderWriting obj);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("registerObject:visibility:")]
+		void RegisterObject (INSItemProviderWriting obj, NSItemProviderRepresentationVisibility visibility);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("registerObjectOfClass:visibility:loadHandler:")]
+		void RegisterObject (INSItemProviderWriting aClass, NSItemProviderRepresentationVisibility visibility, RegisterObjectLoadHandler loadHandler);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("canLoadObjectOfClass:")]
+		bool CanLoadObject (INSItemProviderReading aClass);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("loadObjectOfClass:completionHandler:")]
+		NSProgress LoadObject (INSItemProviderReading aClass, LoadObjectCompletionHandler completionHandler);
+	}
+
+	interface INSItemProviderWriting { }
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface NSItemProviderWriting
+	{
+		//TODO: Where to implement this?
```

Comment:
Please file a bug about this so that we don't forget (and add the bug # in the comment).